pumps/kafka.go

```diff
@@ -104,23 +106,29 @@ func (k *KafkaPump) WriteData(data []interface{}) error {
 		}
 
 		//Transform object to json string
-		json, jsonError := json.Marshal(message)
-		if jsonError != nil {
-			k.log.WithError(jsonError).Error("unable to marshal message")
+		jsBytes, err := json.Marshal(message)
+		if err != nil {
+			k.log.WithError(err).Error("unable to marshal message")
 		}
 
 		//Kafka message structure
 		kafkaMessages[i] = kafka.Message{
 			Time:  time.Now(),
-			Value: json,
+			Value: jsBytes,
 		}
 	}
-	//Send kafka message
-	kafkaError := k.write(kafkaMessages)
-	if kafkaError != nil {
-		k.log.WithError(kafkaError).Error("unable to write message")
-	}
-	k.log.Debug("ElapsedTime in seconds for ", len(data), " records:", time.Now().Sub(startTime))
+
+	// send kafka message in separate go-routine so that will not block the next Redis purge in event of Kafka slowdown.
+	// should only create new go-routine every `config.purge_delay` seconds
+	go func(startTime time.Time, log *logrus.Entry) {
+		err := k.write(kafkaMessages)
+		if err != nil {
+			log.WithError(err).Error("unable to write message")
+		}
+
+		log.Infof("written %d records %s", len(data), time.Now().Sub(startTime).String())
```

Comment:
fixed the message to refer to the time period 
"written %d records in %s seconds"

Reply:
BTW, should we use _k.log.Time_ instead of _time.Now()_ ?

Reply:
actually it's instead of passing the start time. since the log entry is by value

Reply:
actually about _log *logrus.Entry_ - should it be a pointer? it's not in _func (entry Entry) log(level Level, msg string)_

Reply:
1. the `String()` method automatically adds sensible units. So it will log: `written 100000 records 1.2s` or similar
2. I've never used log.Time - I think that referrs to the time the log was written
3. We shouldn't really be timing jobs here anyway, but actually using instrumentation.

```
job.Timing("purge_time_"+pmp.GetName(), time.Since(startTime).Nanoseconds())
```

Reply:
```
// String returns a string representing the duration in the form "72h3m0.5s".
// Leading zero units are omitted. As a special case, durations less than one
// second format use a smaller unit (milli-, micro-, or nanoseconds) to ensure
// that the leading digit is non-zero. The zero duration formats as 0s.
```

Reply:
About entry.Time: check the log function in entry.go

> func (entry Entry) log(level Level, msg string) {
> 	var buffer *bytes.Buffer
> 	entry.Time = time.Now()